### PR TITLE
add an ECS task role to send email using SES and make SMTP vars optional

### PR DESCRIPTION
**Pull request by briskt**
### Added
- Added an ECS task role to the email-service module for sending email using SES.

### Changed
- Made email-service SMTP variables optional.

### CODE REVIEW

File: terraform/031-email-service/main.tf [hunk 1]
```diff
@@ -50,6 +50,52 @@ resource "random_id" "access_token_idsync" {
...
+      Version = "2012-10-17"
+      Statement = [
+        {
+          Sid      = "SendEmail"
+          Effect   = "Allow"
+          Action   = "ses:SendEmail"
+          Resource = "*"
+          Condition = {
+            StringEquals = {
+              "ses:FromAddress" = var.from_email
```
Comment (briskt) [inline terraform/031-email-service/main.tf:91]: Reference: https://docs.aws.amazon.com/ses/latest/dg/sending-authorization-policy-examples.html

Comment (briskt) [inline terraform/031-email-service/main.tf:91]: Reference: https://docs.aws.amazon.com/ses/latest/dg/sending-authorization-policy-examples.html

File: terraform/031-email-service/main.tf [hunk 2]
```diff
@@ -50,6 +50,52 @@ resource "random_id" "access_token_idsync" {
...
+        Effect = "Allow"
+        Principal = {
+          Service = [
+            "ses.amazonaws.com",
+            "ecs-tasks.amazonaws.com",
+          ]
+        }
+        Action = "sts:AssumeRole"
+      }
+    ]
```
Comment (forevermatt) [inline terraform/031-email-service/main.tf:73]: I'm curious: how did you learn/find what needed to be in this policy?

File: terraform/031-email-service/vars.tf [hunk 3]
```diff
@@ -91,20 +91,27 @@ variable "internal_alb_listener_arn" {
 }
 
 variable "mailer_host" {
-  type = string
+  description = "SMTP hostname - if omitted, SES will be used"
```
Comment (forevermatt) [inline terraform/031-email-service/vars.tf:94]: Hooray for in-code documentation of terraform variables! 😁

**Review by forevermatt — APPROVED** [review]

File: terraform/031-email-service/main.tf (hunk 2 above)
Comment (briskt) [reply terraform/031-email-service/main.tf:73]: I did it 2+ years ago, so I don't remember. I did try to relearn it recently, though, which is why I included a link to the AWS documentation.

Comment (briskt) [reply terraform/031-email-service/main.tf:73]: Oh, I guess the AWS link is for the other policy. I think we've used a policy similar to this one in other places. Can't remember where, at the moment.

**Pull request merged**